paymentsheet/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/ach/USBankAccountFormFragment.kt

```diff
@@ -252,16 +266,14 @@ internal class USBankAccountFormFragment : Fragment() {
     ) {
         setContent {
             PaymentsTheme {
-                MandateCollectionScreen(
-                    screenState.bankName,
-                    screenState.displayName,
-                    screenState.last4
-                )
+                MandateCollectionScreen(screenState)
             }
         }
         updatePrimaryButton(
             text = screenState.primaryButtonText,
-            onClick = screenState.primaryButtonOnClick,
+            onClick = {
+                viewModel.handlePrimaryButtonClick(screenState)
```

Comment:
Nice I really like this calling into a viewModel!  Is it possible for the screenState to change between when the primaryButton state is setup to the time that click is called?  If possible it might be better to have state saved in the viewModel.